python/paddle/fluid/optimizer.py

```diff
@@ -291,18 +292,41 @@ def _process_distribute_lookuptable(self, param_grads, loss,
                         outputs={"ParamOut": param_and_grad[0]})
         return new_param_grads, (table_param, table_grad), sgd_op
 
+    def _backward(self,
+                  loss,
+                  startup_program=None,
+                  parameter_list=None,
+                  no_grad_set=None,
+                  callbacks=None):
+        if callbacks is None:
+            callbacks = [error_clip_callback]
+        else:
+            assert (isinstance(callbacks, list))
+            callbacks.append(error_clip_callback)
+        self._backward_called = True
+        self.params_grads = append_backward(loss, parameter_list, no_grad_set,
+                                            callbacks)
+        return self.params_grads
+
     def minimize(self,
                  loss,
                  startup_program=None,
                  parameter_list=None,
-                 no_grad_set=None):
+                 no_grad_set=None,
+                 user_params_grads=None):
         """Add operations to minimize `loss` by updating `parameter_list`.
 
         This method combines interface `append_backward()` and
         `create_optimization_pass()` into one.
         """
-        params_grads = append_backward(loss, parameter_list, no_grad_set,
-                                       [error_clip_callback])
+        if not self._backward_called:
```

Comment:
Is `_backward` interface is public?

Reply:
not recommended for user to call this. only used for mix precision for optimize using fp32.